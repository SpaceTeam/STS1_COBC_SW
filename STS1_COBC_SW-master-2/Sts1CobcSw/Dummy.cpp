#include <Sts1CobcSw/Dummy.hpp>


namespace sts1cobcsw
{
Dummy::Dummy() : name("Sts1CobcSw")
{
}
}
