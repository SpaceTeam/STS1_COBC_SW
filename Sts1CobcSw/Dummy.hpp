#pragma once


#include <etl/string.h>


namespace sts1cobcsw
{
struct Dummy
{
    Dummy();

    static constexpr auto maxNameLength = 42;

    etl::string<maxNameLength> name = "";
};
}
