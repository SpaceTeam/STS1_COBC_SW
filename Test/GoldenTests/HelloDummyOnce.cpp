#include <Sts1CobcSw/Dummy.hpp>

#include <rodos.h>


uint32_t printfMask = 0;

namespace sts1cobcsw
{
class HelloDummyOnce : public StaticThread<>
{
    void run() override
    {
        printfMask = 1;
        auto const dummy = Dummy();

        PRINTF("Hello, %s!\n", dummy.name.data());
        hwResetAndReboot();
    }
};

auto const helloDummyOnce = HelloDummyOnce();
}
