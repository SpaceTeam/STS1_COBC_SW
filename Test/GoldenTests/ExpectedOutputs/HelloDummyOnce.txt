Hello, Sts1CobcSw!
hw_resetAndReboot() -> exit
